### DISCUSSION TITLE
Compilation error

### DISCUSSION THREAD

**Discussion by jtroeth1** [eProsima/Fast-DDS-python#87] [Support]
Hi There,

I am trying to compile FastDDS on Fedora 35, but I get the following error:

`❯ /usr/bin/make
[ 33%] Swig compile fastdds.i for python
/home/jtroeth/software/Fast-DDS-python/fastdds_python/src/swig/fastdds/rtps/attributes/ThreadSettings.i:19: Error: Unable to find 'fastdds/rtps/attributes/ThreadSettings.hpp'
make[2]: *** [src/swig/CMakeFiles/fastdds_python_swig_compilation.dir/build.make:76: src/swig/CMakeFiles/fastdds_python.dir/fastddsPYTHON.stamp] Error 1
make[2]: *** Deleting file 'src/swig/CMakeFiles/fastdds_python.dir/fastddsPYTHON.stamp'
make[1]: *** [CMakeFiles/Makefile2:126: src/swig/CMakeFiles/fastdds_python_swig_compilation.dir/all] Error 2
make: *** [Makefile:146: all] Error 2`

I assume the ThreadSettings.hpp is mean to be generated, but is not?

Versions:
- swig 4.0.2
- Fast CDR: master 71d2cc2
- foonathan_memory_vendor: master 48e5e2c
- FastDDS: master e318c5b99
- FastDDS-python, main or v1.4.0
- asio-devel 1.16.1-4.fc35

Thanks

**Comment by jtroeth1** [comment]
Removing ThreadSettings.i from fastdds_python/src/swig/fastdds.i allows for a successful build.

**Comment by JesusPoderoso** [comment]
Hi @jtroeth1 thanks for using Fast DDS.
I've checked that the Fast DDS commit hash you provided does not include the ThreadSettings feature that Fast DDS python expects. That feature was introduced later.
I have tried to reproduce your issue locally and it works for me.
Please, try to `git pull` all repositories and check again.

**Reply by jtroeth1** [reply]
Thanks for the response @JesusPoderoso .

Sorry which version of FastDDS and FastDDs-python should I be using?

The ThreadSettings.i is in FastDDs-python v1.4.0: https://github.com/eProsima/Fast-DDS-python/blob/v1.4.0/fastdds_python/src/swig/fastdds.i#L144

**Reply by JesusPoderoso** [reply]
I would recommend you @jtroeth1 to use the latest versions released in both cases:
- Fast DDS v2.13.1
- Fast DDS python v1.4.0
The other latest versions are, just in case, the following:
- Fast CDR v2.1.3
- Fast DDS Gen v3.2.1
- Fast DDS Gen / third-party / IDL-parser v3.0.0
- eProsima's fork of Foonathan Memory Vendor v1.3.1

**Reply by jtroeth1** [reply]
Great, thanks for your help @JesusPoderoso !

**Comment by JesusPoderoso** [comment]
Also, I am moving this question to the support discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md).
